app/controllers/stats/data/services_controller.rb

```diff
@@ -77,4 +77,12 @@ def set_source
       render_error "Service not found", :status => :not_found
     end
   end
+
+  def stats_params
+    required = %i[metric_name period since]
+
+    permitted_params = params.permit(required | %i[timezone])
+    permitted_params.require(required)
+    permitted_params
```

Comment:
```suggestion
    params.permit(required | %i[timezone]).require(required)
```
I think this would be equivalent, and more concise

Reply:
It wouldn't be the same because `.require(required)` will return an array of values for the required keys. We would lose other parameters that are not required but still permitted.

Reply:
`.permit` filters only permitted parameters and we want to save that value.
https://github.com/rails/rails/blob/v5.0.7.2/actionpack/lib/action_controller/metal/strong_parameters.rb#L469

`.require` checks the parameters for required keys and returns them. We only want to check their presence.
https://github.com/rails/rails/blob/v5.0.7.2/actionpack/lib/action_controller/metal/strong_parameters.rb#L406

Reply:
@hallelujah this explains your questions as well

Reply:
I see, thanks @jsmadis !